fdbserver/workloads/TenantManagementWorkload.actor.cpp

```diff
@@ -221,11 +234,13 @@ struct TenantManagementWorkload : TestWorkload {
 		if (self->useMetacluster) {
 			ASSERT(g_simulator->extraDatabases.size() == 1);
 			self->dataDb = Database::createSimulatedExtraDatabase(g_simulator->extraDatabases[0], cx->defaultTenant);
+			// wait for tenant mode change on dataDB
+			wait(success(self->waitDataDbTenantModeChange()));
```

Comment:
This would probably be more natural just below line 196 where we are registering the database.

Reply:
Due to `waitDataDbTenantModeChange` uses `self->dataDb` variable, I prefer to put it here after `self->dataDb` is setup

Reply:
Yeah I see what you mean, though this is still perhaps a little unclear because not all of the clients need to do this. Perhaps you could move the creation of `self->dataDb` higher up in this function and then you could do the wait in the appropriate block.